src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/UriExtensions.cs

```diff
@@ -4,11 +4,19 @@
 using System;
 using System.Net;
 using Microsoft.AspNetCore.Razor.Utilities;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
 
 namespace Microsoft.CodeAnalysis.Razor;
 
 internal static class UriExtensions
 {
+    /// <summary>
+    ///  Converts the specified <see cref="Uri"/> into a file path that matches
+    ///  a Roslyn <see cref="TextDocument.FilePath"/>.
+    /// </summary>
+    public static string GetDocumentFilePath(this Uri uri)
```

Comment:
I was expecting this method to only be in remote, so it can't be accidentally used in the Razor LSP, but having "Roslyn" in the comment is probably good enough.

Reply:
When I found the same code in `CSharpTestLspServerHelpers`, I decided that I'd rather just share the method across the code base.

Reply:
Oh I missed that, good call!